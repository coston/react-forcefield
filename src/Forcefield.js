import React from 'react'

class Forcefield extends React.Component {
  constructor(props) {
    super(props)
    this.state = { sealed: true }
    this.unseal = this.unseal.bind(this)
  }

  unseal() {
    this.setState({
      sealed: false,
    })
  }

  render() {
    return (
      <div
        style={{
          position: 'relative',
          minWidth: '100%',
        }}
      >
        {this.props.children}
        {this.state.sealed === true ? (
          <div
            onClick={this.unseal}
            style={{
              position: 'absolute',
              top: '0',
              bottom: '0',
              left: '0',
              right: '0',
              height: '100%',
              width: 'auto',
            }}
          />
        ) : null}
      </div>
    )
  }
}

export default Forcefield
