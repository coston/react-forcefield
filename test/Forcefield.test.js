import React from 'react'
import Forcefield from '../src/Forcefield'

test('Render Table', () => {
  const wrapper = shallow(
    <Forcefield>
    <iframe
      src="http://oira.ua.edu/maps/ache/"
      style={{ border: '1px #eee solid' }}
      name="myiFrame"
      scrolling="no"
      height="800px"
      width="100%"
      allowFullScreen
    />
</Forcefield>
  )
  expect(wrapper).toMatchSnapshot()
})
