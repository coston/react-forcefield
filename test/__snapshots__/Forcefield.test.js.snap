// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Table 1`] = `
<div
  style={
    Object {
      "minWidth": "100%",
      "position": "relative",
    }
  }
>
  <iframe
    allowFullScreen={true}
    height="800px"
    name="myiFrame"
    scrolling="no"
    src="http://oira.ua.edu/maps/ache/"
    style={
      Object {
        "border": "1px #eee solid",
      }
    }
    width="100%"
  />
  <div
    onClick={[Function]}
    style={
      Object {
        "bottom": "0",
        "height": "100%",
        "left": "0",
        "position": "absolute",
        "right": "0",
        "top": "0",
        "width": "auto",
      }
    }
  />
</div>
`;
